stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -414,6 +414,43 @@ function triu(A::Symmetric, k::Integer=0)
     end
 end
 
+function dot(A::Symmetric{T, Matrix{T}}, B::Symmetric{T, Matrix{T}}) where T
+    dotprod = zero(T)
+    m, n = size(A)
+    mB, nB = size(B)
```

Comment:
Both `A` and `B` are square by type, so it would suffice to get `size(_, 1)` and do only one comparison below... and use only `m` and not both `m` and `n` in the loops.